front_end/content.py

```diff
@@ -1072,21 +1074,23 @@ def get_same_suggestion(self, help_request):
                        FROM exercises
                        WHERE exercise_id = ?
                    )'''
-        
+
         matches = self.fetchall(sql, (help_request["course_id"], help_request["exercise_id"]))
         for match in matches:
             if match["output_type"] == "jpg":
                 if match["image_output"] == help_request["image_output"]:
                     return match["suggestion"]
             else:
                 if match["text_output"] == help_request["text_output"]:
-                    return match["suggestion"]            
+                    return match["suggestion"]
 
     def get_exercise_submissions(self, course_id, assignment_id, exercise_id):
         exercise_submissions = []
 
-        sql = '''SELECT s.code, u.user_id, u.name, sc.score, s.passed
+        sql = '''SELECT MAX(s.date), s.code, u.user_id, u.name, sc.score, s.passed, p.name AS partner_name
```

Comment:
This code grabs the most recent record and partner name but we still have to address being able to parse through past submissions to see whether a student has ever passed an assignment (more details in Trello card)